testing/E2E_Test_Update.md

```diff
@@ -0,0 +1,163 @@
+device and emulators have been created.  As of yet no real devices have been 
+attached to the slave to provide the the ability to run the tests on real 
+devices. The only thing that needs to be checked is the compatability of the
+Chromedriver and the emulators installed on the macOS slave.
+
+## Making a Solid Test Suite
+
+Where possible, the use of preferred selectors such as AccessibilityId, Id, name etc., 
+placed inside the applications on relevant UI elements would be preferable, as it would 
+eliminate the risks of minor changes in the development of the application, such as moving 
+the order or position of elements on a page or changing text values, requiring an update 
+in the UI tests. 
+
+The use of common design in application development coupled with the development of the 
+suite of tests in a single repository that contains smart abstractions can be major factors 
+in the reduction of the amount of code that will be written and in turn that needs to be 
+maintained.  Common design of application means that only a single test needs to be written 
+to cover all platforms that need to be covered if abstractions such as page objects and 
+data objects are used to remove selectors and expected data from the the tests themselves.  
+These abstractions can then be passed into the tests for multiple platforms.  Indeed, the 
+Java-client has support for cross-platform page objects through the use of special 
+annotations.
+
+## Updated Proposals
+
+A decision should first be made if prefferred selectors are to be implmented in the mobile
+showcase applications.  The effects of this decision are shown below:
+
+#### 1). Using Prefferred Selectors in Applications
+
+* Tests are likely to be less brittle, therefore less time should be required on 
+maintainence
+* Any client can be used provided it has the ability to handle redirects to the browser
+
+
+#### 2). Not Using Prefferred Selectors in Applications
+
+* Tests are likely to be more brittle, therefore more time required to maintain them
+* The Java-client should be the preferred choice of client as it has more stragtegies to
+target locators than other libraries
+
+### Possible Test Suite Implementations
+
+1. A nodejs implmentation with webdriverio - This should be considered if there is a 
+decision to use preferrd selecotrs is to use them in the showcase applications as there 
+is a lot of experience using nodejs within the teams.
+
+1. A Java-client implementation or the Akow library - Should be considered if the decision
+on using prefered selectors is not to use them. This will allow for more locator strategies 
+thatcan be used to create the tests.  The downside of this approach could be the upkeep of 
+the Akow library. 
+
+## Test Writing Time Estimation
+
+### Writing New tests
+
+A new test should take approximately 3-4 hours to complete regardless of the language of 
+implementation.  Appium Inspector can be used to find the selectors and also generate an 
```

Comment:
@twnolan there is an unnecessary double space at the beginning of the sentence (same applies to the sentence in the next line)